src/monodroid/jni/monodroid-glue.c

```diff
@@ -3361,13 +3361,29 @@ set_profile_options (JNIEnv *env)
 	if (!output) {
 		const char* col = strchr (value, ':');
 		char *ovalue;
+		char *extension;
 
-		output = path_combine (override_dirs [0], "profile.mlpd");
+		if (col && !strncmp (value, "log:", 4)
+		    || !strncmp (value, "log", 3))
+			extension = "mlpd";
+		else {
+			int len = col ? col - value : strlen (value) + 1;
+			extension = xmalloc (len);
+			strncpy (extension, value, len);
+			extension [len] = 0;
```

Comment:
Shouldn't this be `extension[len-1]=0`?

For example, if `len` is 4, then `extension` refers to 4 bytes of memory, while `extension[len]` will be *index 4* will be `extension[4]`, which isn't valid.

Reply:
Good catch, thanks. Too much copy'n'paste :-)